project/src/main/java/com/google/sps/data/Places.java

```diff
@@ -38,5 +44,38 @@
     return ImmutableList.copyOf(mutablePlaces);
   }
 
+  /**
+   * filters the given list of places according to the given parameters.
+   * @param places the list we want to sort.
+   * @param minRating The minimal rating we want to limit the list by. A number between 1 and 5.
+   * @param filterIfNoWebsite Specifies whether we should filter if there is no available website
+   *        for the restaurant.
+   * @param filterBranchesOfSamePlace Specifies whether we should remove different branches of the
+   *        same place from the results. Places are considered the same if they have the same name.
+   * @return the filtered list.
+   */
+  public static ImmutableList<Place> filter(ImmutableList<Place> places, int minRating, Boolean
+      filterIfNoWebsite, Boolean filterBranchesOfSamePlace) {
+    //TODO(M1): also filter by place's status when the attribute is added
+    ImmutableList<Place> result = ImmutableList.copyOf(
+        places.stream()
+            .filter(place -> place.rating() >= minRating)
+            .filter(filterIfNoWebsite
+                ? place -> !Strings.isNullOrEmpty(place.websiteUrl())
+                : place -> true)
+            .filter(filterBranchesOfSamePlace
+                ? distinctByKey(p -> p.name())
```

Comment:
Sometimes "place" is used and sometimes "p" so I think to be consistent maybe keep using "place"

Reply:
Great point, changed